packages/foam-vscode/src/core/services/markdown-link.ts

```diff
@@ -38,29 +38,41 @@ export abstract class MarkdownLink {
 
   public static createUpdateLinkEdit(
     link: ResourceLink,
-    delta: { target?: string; section?: string; alias?: string }
+    delta: {
+      target?: string;
+      section?: string;
+      alias?: string;
+      type?: 'wikilink' | 'link';
+      isEmbed?: boolean;
+    }
   ) {
     const { target, section, alias } = MarkdownLink.analyzeLink(link);
     const newTarget = delta.target ?? target;
     const newSection = delta.section ?? section ?? '';
     const newAlias = delta.alias ?? alias ?? '';
     const sectionDivider = newSection ? '#' : '';
     const aliasDivider = newAlias ? '|' : '';
-    const embed = link.isEmbed ? '!' : '';
-    if (link.type === 'wikilink') {
+    const embed = delta.isEmbed ?? link.isEmbed ? '!' : '';
+    const type = delta.type ?? link.type;
+    if (type === 'wikilink') {
       return {
         newText: `${embed}[[${newTarget}${sectionDivider}${newSection}${aliasDivider}${newAlias}]]`,
         range: link.range,
       };
     }
-    if (link.type === 'link') {
+    if (type === 'link') {
+      const defaultAlias = () => {
```

Comment:
out of curiosity, why are you doing this as a function instead of a variable?

Reply:
Lazily evaluating the value of defaultAlias, it's kind of like moving useAngles to `type == 'link'` branch.